### PR TITLE
Ensure latest versions of external plugins still work

### LINKED ISSUE #1762: Update all plugin versions to latest for third-party smoke tests
### Expected Behavior

All external plugins pass the smoke test (module `smoke-test`) with their latest version. Inform plugin authors about potential breakages.

### Current Behavior

Smoke tests run with outdated plugin versions.

### Context

Gradle 4.0 will introduce breaking changes. Make sure that plugins are still buildable with the latest code changes.

### PR DESCRIPTION
**Pull request by w25r**
### Context
Fixes #1762

[CI Results](https://builds.gradle.org/project.html?projectId=Gradle_Branches_Checkpoints&branch_Gradle_Branches_Checkpoints=wl-update-plugins-in-smoketests)

### Contributor Checklist
- [x] Ensure that tests pass locally

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status

### CODE REVIEW

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy [hunk 1]
```diff
@@ -148,7 +155,7 @@ class ThirdPartyPluginsSmokeTest extends AbstractSmokeTest {
                     mavenCentral()
                 }
                 dependencies {
-                    classpath('org.springframework.boot:spring-boot-gradle-plugin:1.4.0.RELEASE')
+                    classpath('org.springframework.boot:spring-boot-gradle-plugin:1.5.2.RELEASE')
```
Comment (bmuschko) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy]: Maybe we should try out https://spring.io/blog/2017/04/05/spring-boot-s-new-gradle-plugin as well. We could potentially had a Spock `where` clause to pass in different versions.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy [hunk 2]
```diff
@@ -269,7 +276,7 @@ class ThirdPartyPluginsSmokeTest extends AbstractSmokeTest {
         given:
         buildFile << """
             plugins {
-                id "org.xtext.xtend" version "1.0.5"
+                id "org.xtext.xtend" version "1.0.17"
```
Comment (bmuschko) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy]: We should also try to use the latest xtend lib: https://bintray.com/bintray/jcenter/org.eclipse.xtend%3Aorg.eclipse.xtend.lib/2.11.0

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy [hunk 3]
```diff
@@ -23,7 +23,7 @@ class NebulaPluginsSmokeTest extends AbstractSmokeTest {
         buildFile << """
             plugins {
                 id "java"
-                id "nebula.dependency-recommender" version "3.3.0"
+                id "nebula.dependency-recommender" version "4.1.2"
             }
```
Comment (bmuschko) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy]: There're also a new versions for other Nebula plugins:

- `nebula.plugin-plugin` -> 5.6.0: https://plugins.gradle.org/plugin/nebula.plugin-plugin
- `nebula.lint` -> 7.3.5: https://plugins.gradle.org/plugin/nebula.lint
- `nebula.dependency-lock` -> 4.9.4:https://plugins.gradle.org/plugin/nebula.dependency-lock

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy (hunk 2 above)
Comment (w25r) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy]: The latest documentation still refers to that version of the xtend lib... but it also refers to 1.0.0 of the plugin.  Should I go ahead and update it? 

http://xtext.github.io/xtext-gradle-plugin/xtend.html

Comment (bmuschko) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy]: Their documentation might just be outdated. I'd upgrade to the latest version. @oehme might know more as he was involved in the implementation of the plugin I believe.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy (hunk 3 above)
Comment (w25r) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy]: Sorry I missed those!

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy (hunk 1 above)
Comment (w25r) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy]: The new plugin is still in snapshot, so we'll wait to add it at this time.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy (hunk 2 above)
Comment (w25r) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/ThirdPartyPluginsSmokeTest.groovy]: Done.

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy (hunk 3 above)
Comment (w25r) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/NebulaPluginsSmokeTest.groovy]: Done.

**Pull request merged**